src/Libraries/PythonNodeModelsWpf/IronPythonCompletionProvider.cs

```diff
@@ -64,84 +66,166 @@ public ScriptScope Scope
+            // Determine if the Revit API is available in the given context
+            if (assemblies.Any(x => x.GetName().Name == "RevitAPI"))
             {
+                // Try to load Revit Type for autocomplete
                 try
                 {
-                    scope.Engine.CreateScriptSourceFromString("import clr\n", SourceCodeKind.Statements).Execute(scope);
-
                     var revitImports =
                         "clr.AddReference('RevitAPI')\nclr.AddReference('RevitAPIUI')\nfrom Autodesk.Revit.DB import *\nimport Autodesk\n";
 
-                    scope.Engine.CreateScriptSourceFromString(revitImports, SourceCodeKind.Statements).Execute(scope);
+                    engine.CreateScriptSourceFromString(revitImports, SourceCodeKind.Statements).Execute(scope);
+                    clrModules.Add("RevitAPI");
+                    clrModules.Add("RevitAPIUI");
                 }
                 catch
                 {
-                    Log("Failed to load Revit types for autocomplete.  Python autocomplete will not see Autodesk namespace types.");
+                    Log("Failed to load Revit types for autocomplete. Python autocomplete will not see Autodesk namespace types.");
                 }
             }
 
-            if (assemblies.Any(x => x.FullName.Contains("ProtoGeometry")))
+            // Determine if the ProtoGeometry is available in the given context
+            if (assemblies.Any(x => x.GetName().Name == "ProtoGeometry"))
             {
+                // Try to load ProtoGeometry Type for autocomplete
                 try
                 {
-                    scope.Engine.CreateScriptSourceFromString("import clr\n", SourceCodeKind.Statements).Execute(scope);
-
                     var libGImports =
-                        "import clr\nclr.AddReference('ProtoGeometry')\nfrom Autodesk.DesignScript.Geometry import *\n";
+                        "clr.AddReference('ProtoGeometry')\nfrom Autodesk.DesignScript.Geometry import *\n";
 
-                    scope.Engine.CreateScriptSourceFromString(libGImports, SourceCodeKind.Statements).Execute(scope);
+                    engine.CreateScriptSourceFromString(libGImports, SourceCodeKind.Statements).Execute(scope);
+                    clrModules.Add("ProtoGeometry");
                 }
                 catch (Exception e)
                 {
                     Log(e.ToString());
-                    Log("Failed to load ProtoGeometry types for autocomplete.  Python autocomplete will not see Autodesk namespace types.");
+                    Log("Failed to load ProtoGeometry types for autocomplete. Python autocomplete will not see Autodesk namespace types.");
                 }
             }
 
+            // Determine if the Python Standard Library is available in the given context
+            if(!String.IsNullOrEmpty(dynamoCoreDir))
+            {
+                pythonLibDir = dynamoCoreDir + @"\IronPython.StdLib.2.7.8";
```

Comment:
@alfarok don't do this - instead use System.Path.Combine - it will work cross platform.